### PR TITLE
Shot animation

### PR DESCRIPTION
**Pull request by vlad2876**

### CODE REVIEW

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 1]
```diff
@@ -9,11 +9,31 @@ export class SeaAreaComponent implements OnInit {
...
+      let activeRowIndex: number = this.rowsCount;
+      const shotInterval = setInterval(() => {
+        activeRowIndex--;
+        this.seaAreaCells[activeRowIndex][this.activeColumnIndex].active = true;
+        if (activeRowIndex < 9) {
+          this.seaAreaCells[activeRowIndex + 1][this.activeColumnIndex].active = false;
+        }
+        if (activeRowIndex === 0) {
+          clearInterval(shotInterval);
+          setTimeout(() => {
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:29]: Please refactor this code in the way not using setTimeout. setTimeout adds excess complexity here. It's enough to use just setInterval here.

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 2]
```diff
@@ -9,11 +9,31 @@ export class SeaAreaComponent implements OnInit {
...
   columnsCount = 15;
 
-  readonly activeRowIndex: number = 9;
-  readonly activeColumnIndex: number = 0;
-
   seaAreaCells: SeaAreaCell[][] = [];
 
+  readonly activeColumnIndex: number = 7;
+
+
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:16]: Please remove the extra line here. Please add a eslint rule to forbid using more than one empty line in a row.

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 3]
```diff
@@ -9,11 +9,31 @@ export class SeaAreaComponent implements OnInit {
...
+  @HostListener('window:keyup', ['$event'])
+  shotAnimation(event: { code: string }) {
+    if (event.code === 'Space') {
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:19]: Why not use the enum value here?

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 4]
```diff
@@ -9,11 +9,31 @@ export class SeaAreaComponent implements OnInit {
...
+      let activeRowIndex: number = this.rowsCount;
+      const shotInterval = setInterval(() => {
+        activeRowIndex--;
+        this.seaAreaCells[activeRowIndex][this.activeColumnIndex].active = true;
+        if (activeRowIndex < 9) {
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:24]: Magic number

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 5]
```diff
@@ -9,11 +9,31 @@ export class SeaAreaComponent implements OnInit {
...
+          this.seaAreaCells[activeRowIndex + 1][this.activeColumnIndex].active = false;
+        }
+        if (activeRowIndex === 0) {
+          clearInterval(shotInterval);
+          setTimeout(() => {
+            this.seaAreaCells[activeRowIndex][this.activeColumnIndex].active = false;
+          }, 120);
+        }
+      }, 120);
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:33]: Magic number

**Comment by Zmitserkos** [comment]
The dog pic is really cute, but is it related to the PR somehow???

File: src/app/components/home/game-container/sea-area/sea-area.component.ts (hunk 2 above)
Comment (vlad2876) [reply src/app/components/home/game-container/sea-area/sea-area.component.ts]: Fixed

File: src/app/components/home/game-container/sea-area/sea-area.component.ts (hunk 3 above)
Comment (vlad2876) [reply src/app/components/home/game-container/sea-area/sea-area.component.ts]: Fixed

File: src/app/components/home/game-container/sea-area/sea-area.component.ts (hunk 4 above)
Comment (vlad2876) [reply src/app/components/home/game-container/sea-area/sea-area.component.ts]: Fixed

File: src/app/components/home/game-container/sea-area/sea-area.component.ts (hunk 5 above)
Comment (vlad2876) [reply src/app/components/home/game-container/sea-area/sea-area.component.ts]: Fixed

**Comment by vlad2876** [comment]
> 

What picture are we talking about?

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 6]
```diff
@@ -8,12 +9,33 @@ import {Component, OnInit} from '@angular/core';
 export class SeaAreaComponent implements OnInit {
   rowsCount = 10;
   columnsCount = 15;
-
-  readonly activeRowIndex: number = 9;
-  readonly activeColumnIndex: number = 0;
+  shotSpeed = 120;
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:12]: Please come up with a better naming. This property doesn't really mean speed.

File: src/app/components/home/game-container/sea-area/sea-area.component.ts [hunk 7]
```diff
@@ -8,12 +9,33 @@ import {Component, OnInit} from '@angular/core';
...
+      let activeRowIndex: number = this.rowsCount;
+      const shotInterval = setInterval(() => {
+        activeRowIndex--;
+        this.seaAreaCells[activeRowIndex][this.activeColumnIndex].active = true;
+        if (activeRowIndex < this.rowsCount - 1) {
+          this.seaAreaCells[activeRowIndex + 1][this.activeColumnIndex].active = false;
+        }
+        if (activeRowIndex === 0) {
+          clearInterval(shotInterval);
+          const clearUpperCell = setInterval(() => {
```
Comment (Zmitserkos) [inline src/app/components/home/game-container/sea-area/sea-area.component.ts:30]: Oh no ;)

Comment (vlad2876) [reply src/app/components/home/game-container/sea-area/sea-area.component.ts]: Fixed

File: src/app/components/home/game-container/sea-area/sea-area.component.ts (hunk 6 above)
Comment (vlad2876) [reply src/app/components/home/game-container/sea-area/sea-area.component.ts]: Fixed

**Pull request merged**